kpi/deployment_backends/kobocat_backend.py

```diff
@@ -449,132 +478,80 @@ def get_submission_validation_status_url(self, submission_pk):
-    def _get_deployment(self, request):
-        '''
-        Presupposing the use of `NestedViewSetMixin`, return the deployment for
-        the asset specified by the KPI request
-        '''
-        asset_uid = self.get_parents_query_dict()['asset']
-        asset = get_object_or_404(self.parent_model, uid=asset_uid)
-        if not asset.has_deployment:
-            raise serializers.ValidationError(
-                _('The specified asset has not been deployed'))
-        if not isinstance(asset.deployment, KobocatDeploymentBackend):
-            raise NotImplementedError(
-                'This viewset can only be used with the KoBoCAT deployment '
-                'backend')
-        return asset.deployment
-
-    @staticmethod
-    def _kobocat_proxy_request(kpi_request, kc_request):
-        '''
-        Send `kc_request`, which must specify `method` and `url` at a minimum.
-        If `kpi_request`, i.e. the incoming request to be proxied, is
-        authenticated, logged-in user's API token will be added to
-        `kc_request.headers`
-        '''
-        user = kpi_request.user
-        if not user.is_anonymous() and user.pk != settings.ANONYMOUS_USER_ID:
-            token, created = Token.objects.get_or_create(user=user)
-            kc_request.headers['Authorization'] = 'Token %s' % token.key
-        session = requests.Session()
-        return session.send(kc_request.prepare())
-
-    @staticmethod
-    def _requests_response_to_django_response(requests_response):
-        '''
-        Convert a `requests.models.Response` into a `django.http.HttpResponse`
-        '''
-        HEADERS_TO_COPY = ('Content-Type', 'Content-Language')
-        django_response = HttpResponse()
-        for header in HEADERS_TO_COPY:
-            try:
-                django_response[header] = requests_response.headers[header]
-            except KeyError:
-                continue
-        django_response.status_code = requests_response.status_code
-        django_response.write(requests_response.content)
-        return django_response
-
-
-    def list(self, kpi_request, *args, **kwargs):
-        return self.retrieve(kpi_request, None, *args, **kwargs)
-
-    def retrieve(self, kpi_request, pk, *args, **kwargs):
-        deployment = self._get_deployment(kpi_request)
-        if pk is None:
-            kc_url = deployment.submission_list_url
+    def get_submissions(self, format_type=INSTANCE_FORMAT_TYPE_JSON, instances_uuids=[]):
+        """
+        Retreives submissions through Postgres or Mongo depending on `format_type`.
+        It can be filtered on instances uuids.
+        `uuid` is used instead of `id` because `id` is not available in ReadOnlyInstance model
+
+        :param format_type: str. INSTANCE_FORMAT_TYPE_JSON|INSTANCE_FORMAT_TYPE_XML
+        :param instances_uuids: list. Optional
+        :return: list: mixed
+        """
+        submissions = []
+        getter = getattr(self, "_{}__get_submissions_in_{}".format(
```

Comment:
:sparkles: So fancy, I'm almost afraid of it.

Reply:
Don't be scared, it's like magic :-)